ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java

```diff
@@ -211,54 +210,46 @@ public ProtocolSchedule createProtocolSchedule() {
-                    BuilderMapEntry::getBlock,
-                    b -> b,
-                    (existing, replacement) -> replacement,
-                    TreeMap::new));
-
-    return builders;
+    return Lists.newArrayList(
+            create(OptionalLong.of(0), specFactory.frontierDefinition()),
+            create(config.getHomesteadBlockNumber(), specFactory.homesteadDefinition()),
+            create(
+                config.getTangerineWhistleBlockNumber(), specFactory.tangerineWhistleDefinition()),
+            create(config.getSpuriousDragonBlockNumber(), specFactory.spuriousDragonDefinition()),
+            create(config.getByzantiumBlockNumber(), specFactory.byzantiumDefinition()),
+            create(config.getConstantinopleBlockNumber(), specFactory.constantinopleDefinition()),
+            create(config.getPetersburgBlockNumber(), specFactory.petersburgDefinition()),
+            create(config.getIstanbulBlockNumber(), specFactory.istanbulDefinition()),
+            create(config.getMuirGlacierBlockNumber(), specFactory.muirGlacierDefinition()),
+            create(config.getBerlinBlockNumber(), specFactory.berlinDefinition()),
+            create(config.getLondonBlockNumber(), specFactory.londonDefinition(config)),
+            create(config.getArrowGlacierBlockNumber(), specFactory.arrowGlacierDefinition(config)),
+            create(config.getGrayGlacierBlockNumber(), specFactory.grayGlacierDefinition(config)),
+            create(config.getMergeNetSplitBlockNumber(), specFactory.parisDefinition(config)),
+            create(config.getShandongBlockNumber(), specFactory.shandongDefinition(config)),
+            // Classic Milestones
+            create(config.getEcip1015BlockNumber(), specFactory.tangerineWhistleDefinition()),
+            create(config.getDieHardBlockNumber(), specFactory.dieHardDefinition()),
+            create(config.getGothamBlockNumber(), specFactory.gothamDefinition()),
+            create(
+                config.getDefuseDifficultyBombBlockNumber(),
+                specFactory.defuseDifficultyBombDefinition()),
+            create(config.getAtlantisBlockNumber(), specFactory.atlantisDefinition()),
+            create(config.getAghartaBlockNumber(), specFactory.aghartaDefinition()),
+            create(config.getPhoenixBlockNumber(), specFactory.phoenixDefinition()),
+            create(config.getThanosBlockNumber(), specFactory.thanosDefinition()),
+            create(config.getMagnetoBlockNumber(), specFactory.magnetoDefinition()),
+            create(config.getMystiqueBlockNumber(), specFactory.mystiqueDefinition()),
+            create(config.getEcip1049BlockNumber(), specFactory.ecip1049Definition()))
+        .stream()
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(
+            Collectors.toMap(
+                BuilderMapEntry::getBlock,
+                b -> b,
+                (existing, replacement) -> replacement,
+                TreeMap::new));
```

Comment:
If we are changing this snippet I suggest to remove the usage of `c.g.c.c.Lists#newArrayList` too
```suggestion
    final TreeMap<Long, BuilderMapEntry> builders = new TreeMap<>();
    create(OptionalLong.of(0), specFactory.frontierDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getHomesteadBlockNumber(), specFactory.homesteadDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getTangerineWhistleBlockNumber(), specFactory.tangerineWhistleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getSpuriousDragonBlockNumber(), specFactory.spuriousDragonDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getByzantiumBlockNumber(), specFactory.byzantiumDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getConstantinopleBlockNumber(), specFactory.constantinopleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getPetersburgBlockNumber(), specFactory.petersburgDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getIstanbulBlockNumber(), specFactory.istanbulDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMuirGlacierBlockNumber(), specFactory.muirGlacierDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getBerlinBlockNumber(), specFactory.berlinDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getLondonBlockNumber(), specFactory.londonDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getArrowGlacierBlockNumber(), specFactory.arrowGlacierDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getGrayGlacierBlockNumber(), specFactory.grayGlacierDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMergeNetSplitBlockNumber(), specFactory.parisDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    // Classic Milestones
    create(config.getEcip1015BlockNumber(), specFactory.tangerineWhistleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getDieHardBlockNumber(), specFactory.dieHardDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getGothamBlockNumber(), specFactory.gothamDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(
            config.getDefuseDifficultyBombBlockNumber(),
            specFactory.defuseDifficultyBombDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getAtlantisBlockNumber(), specFactory.atlantisDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getAghartaBlockNumber(), specFactory.aghartaDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getPhoenixBlockNumber(), specFactory.phoenixDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getThanosBlockNumber(), specFactory.thanosDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMagnetoBlockNumber(), specFactory.magnetoDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMystiqueBlockNumber(), specFactory.mystiqueDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getEcip1049BlockNumber(), specFactory.ecip1049Definition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    return builders;

```

Reply:
Done, but slightly differently.  Using `Stream.of`